### PR TITLE
refactor: unify block1 and block2 terminology

### PR DESCRIPTION
**Pull request by JKRhb**
So far both the data structures as well as the terminology in block1 and block2 block state options differed. In preparation for more consistent typing this PR applies some more cleanup to the project.

### CODE REVIEW

File: lib/block.js [hunk 1]
```diff
@@ -7,35 +7,39 @@
...
 
 const TwoPowTwenty = 1048575
+
 /**
  *
- * @param {Number} sequenceNumber The block sequence number.
- * @param {Boolean} moreBlocks If there are more blocks to follow
- * @param {Number} blockSize
+ * @param numOrBlockState The block sequence number or a block state object.
+ * @param more Can indicate if more blocks are to follow.
```
Comment (Apollon77) [inline lib/block.js:14]: "if" somehow indicated that it is a boolean ... but in fact it is a nunber?

File: lib/block.js [hunk 2]
```diff
@@ -7,35 +7,39 @@
...
+    if (typeof numOrBlockState === 'object') {
+        num = numOrBlockState.num
+        more = numOrBlockState.more
+        size = numOrBlockState.size
+    } else {
+        num = numOrBlockState
     }
 
-    if (sequenceNumber == null || blockSize == null || sequenceNumber == null) {
+    if (num == null || more == null || size == null) {
```
Comment (Apollon77) [inline lib/block.js:27]: if you want to make sure these are numbers maybe check that instead of == null

File: lib/block.js (hunk 1 above)
Comment (Apollon77) [reply lib/block.js:14]: and if number with only 0/1 why not boolean?

Comment (JKRhb) [reply lib/block.js:14]: I think I changed it to a boolean first (there was some inconsistency as the block1 implementation used numbers, if I recall correctly) but then used 0/1 instead as the `more` flag can have different meanings depending on the context (see https://datatracker.ietf.org/doc/html/rfc7959#section-2.2). Therefore the "Can indicate". But I guess we could also use boolean values here.

File: lib/block.js (hunk 2 above)
Comment (JKRhb) [reply lib/block.js:27]: Oh, good point!

File: lib/block.js (hunk 1 above)
Comment (Apollon77) [reply lib/block.js:14]: Ok if specs have it as number we better stay near to the specs so maybe just change the text maybe to indicate that

**Pull request merged**